app.py

```diff
@@ -122,9 +122,17 @@ def home_page():
 @app.get('/profile-page')
 @other_methods.check_user
 def profile():
-    if 'user_id' not in session:
-        return redirect(url_for('sign_in'))
-    return render_template('profile.html')
+    current_user = database_methods.get_user_by_id(session['user_id'])
+    user_posts = database_methods.get_posts_by_user_id(session['user_id'])
+
+    return render_template('profile.html', current_user=current_user, user_posts=user_posts)
+
```

Comment:
Relating to the comments I made with the database_methods.py, I think here because you are passing in the "current_user" into the template, you can just add the first and last names into the method ("get_user_by_id") so that it can grab them for you without you having to make new methods. Same thing with concentration.

Or you can just use the method that I suggested to make it easier for yourself, because then you will know what is inside the method instead of having to go back and forth between the files to make sure that "get_user_by_id" has the information you need.